rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -511,11 +511,56 @@ static void dds_listener_callback(dds_entity_t entity, void * arg)
 MAKE_DDS_EVENT_CALLBACK_FN(requested_deadline_missed, REQUESTED_DEADLINE_MISSED)
 MAKE_DDS_EVENT_CALLBACK_FN(liveliness_lost, LIVELINESS_LOST)
 MAKE_DDS_EVENT_CALLBACK_FN(offered_deadline_missed, OFFERED_DEADLINE_MISSED)
-MAKE_DDS_EVENT_CALLBACK_FN(requested_incompatible_qos, REQUESTED_INCOMPATIBLE_QOS)
 MAKE_DDS_EVENT_CALLBACK_FN(sample_lost, SAMPLE_LOST)
-MAKE_DDS_EVENT_CALLBACK_FN(offered_incompatible_qos, OFFERED_INCOMPATIBLE_QOS)
 MAKE_DDS_EVENT_CALLBACK_FN(liveliness_changed, LIVELINESS_CHANGED)
-MAKE_DDS_EVENT_CALLBACK_FN(inconsistent_topic, INCONSISTENT_TOPIC)
+
+/**
+ * Because the inconsistent topic is not handled correctly in CycloneDDS,
```

Comment:
I disagree with the definitive character of this statement *in this context*: I don't believe that it is sufficiently clear what "correct" handling of the "inconsistent topic" notification is to make this statement in the context of detecting the existence of multiple definitions for a topic T in the system with different types.

What I do accept without reservation is that it is a bug that Cyclone fails to raise an "inconsistent topic" notification when topic discovery is enabled while there definitions of T with differing QoS settings.

But this is too strong for the context.

```suggestion
 * Because the inconsistent topic is not raised by CycloneDDS when a reader/writer fail to match because of differing type definitions
```

Reply:
@eboasson Thanks for the clarification. For what it's worth, I did not mean to knock CycloneDDS here, I was just being lazy in my comment. "Not handled correctly" was a poor way to phrase this.